### ISSUE TITLE
Panic when MessagePart Content-Disposition is Empty

### ISSUE THREAD

**Issue by PapayaJackal** [stalwartlabs/mail-parser#63]
I am trying to parse a real-life example of an email where one of the MIME parts looks as follows:

```
--boundary
Content-Type: image/png
Content-Transfer-Encoding: base64
Content-Disposition: ;
        filename="image001.png"
Content-ID: <image001.png@[...]>

[...]
```

This results in a panic:

```
thread 'main' panicked at 'HeaderValue::content_type called on non-ContentType: Empty', /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mail-parser-0.9.0/src/core/header.rs:191:18
stack backtrace:
   0: rust_begin_unwind
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/panicking.rs:67:14
   2: mail_parser::core::header::<impl mail_parser::HeaderValue>::content_type
             at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mail-parser-0.9.0/src/core/header.rs:191:18
   3: mail_parser::parsers::message::<impl mail_parser::MessageParser>::parse_::{{closure}}
             at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mail-parser-0.9.0/src/parsers/message.rs:275:52
   4: core::option::Option<T>::map_or_else
             at /rustc/5680fa18feaa87f3ff04063800aec256c3d4b4be/library/core/src/option.rs:1179:24
   5: mail_parser::parsers::message::<impl mail_parser::MessageParser>::parse_
             at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mail-parser-0.9.0/src/parsers/message.rs:273:24
   6: mail_parser::parsers::message::<impl mail_parser::MessageParser>::parse
             at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/mail-parser-0.9.0/src/parsers/message.rs:117:9
[...]
```

Changing the `Content-Disposition` header to `inline; filename="image001.png"` or `attachment; filename="image001.png"` fixes the panic.

**Comment by mdecimus** [comment]
I'll check it out. Can you post the full message removing any private details?

**Comment by mdecimus** [comment]
If possible please email it to hello@stalw.art

**Comment by mdecimus** [comment]
Actually, it is not needed. I was able to reproduce it.

**Comment by mdecimus** [comment]
Thanks for reporting this, it has now been fixed in `v0.9.1`.
